@xen-orchestra/web-core/lib/components/StatusPill.vue

```diff
@@ -0,0 +1,29 @@
+<template>
+  <UiIcon :color :icon class="status-pill" />
+</template>
+
+<script lang="ts" setup>
+import UiIcon from '@core/components/icon/UiIcon.vue'
+import { computed } from 'vue'
+import { faCheckCircle, faCircleMinus, faCircleXmark } from '@fortawesome/free-solid-svg-icons'
+import type { Color } from '@core/types/color.type'
+
+const props = defineProps<{
+  state: 'success' | 'partial' | 'failure'
+}>()
+
+const states = {
+  success: { icon: faCheckCircle, color: 'success' },
+  partial: { icon: faCircleMinus, color: 'warning' },
+  failure: { icon: faCircleXmark, color: 'error' },
+}
+
+const icon = computed(() => states[props.state].icon)
+const color = computed(() => states[props.state].color as Color)
```

Comment:
We should use strong typing to prevent potential errors.

In this case:
1. If we add a type of `state` (line 12) without adding it to `states` object => no error
2. If we make a typo for the color property (ie. `waning`) in the `states` object => no error
3. `as Color` needs to be forced on line 22, again not complaining about a bad color name

Here is my suggestion:

```suggestion
type Props = {
  state: 'success' | 'partial' | 'failure'
}

const props = defineProps<Props>()

const states: Record<Props['state'], { icon: IconDefinition; color: Color }> = {
  success: { icon: faCheckCircle, color: 'success' },
  partial: { icon: faCircleMinus, color: 'warning' },
  failure: { icon: faCircleXmark, color: 'error' },
}

const icon = computed(() => states[props.state].icon)
const color = computed(() => states[props.state].color)
```

Now:
- if a `props.state` is not handled by `states` => error
- if a `states.color` or a `states.icon` is invalid => error
- it's no longer needed to force the `color` computed type